Create the C modules section (#1195)
MicroEJ/docs: VEEPortingGuide/appendix/cmodules.rst

.. _CMODULES-CHAPTER:

======================
C Modules Installation
======================

This section describes how to install a C module on your VEE Port, depending on your SDK version the first steps will vary.

Fetching the module source files
++++++++++++++++++++++++++++++++

.. tabs::

   .. tab:: SDK 5

      Under the SDK 5 follow these steps:

      - Go to the location of your C module, for example the C module for microUI 14.2.0 is located `here <https://repository.microej.com/modules/com/microej/clibrary/llimpl/microui/14.2.0/>`_.
      - Get the dependency declaration, for the previous example this would be: ``<dependency org="com.microej.clibrary.llimpl" name="microui" rev="14.2.0" />``.
      - Add it inside the block ``<dependencies>`` from the file ``module.ivy`` of your ``*-configuration`` project.
      - Remove the ``.properties`` files in the folder ``*-bsp/projects/microej``.

>       - Build the module ``*-configuration/``.

Reviewer: ```suggestion
      - Rebuild the VEE Port: in the SDK 5 Project Explorer, right click on the VEE Port module ``*-configuration > build module``.
```

   .. tab:: SDK 6

>       Under the SDK 6 follow these steps:

Reviewer: ```suggestion
      Under the SDK 6, follow these steps:
```

      - Go to the location of your C module, for example the C module for microUI 14.2.0 is located `here <https://repository.microej.com/modules/com/microej/clibrary/llimpl/microui/14.2.0/>`_.

>       - Download the archive file ``.cco``.

Reviewer: ```suggestion
      - Download the archive file with the ``.cco`` extension.
```

      - Unzip the content of this file.
      - The source files are located in the folder ``bsp/``.

>       - Copy the content of this folder into your VEE Port BSP.

Reviewer: ```suggestion
      - Copy the **content** of the ``bsp/`` folder into your VEE Port at the path ``bsp/vee/port/``.
```

> Configure and Build

Reviewer: ```suggestion
C module configuration and firmware build
```

Optional suggestion, to be more explicit on the content of this section.

+++++++++++++++++++

> - Update the build of your BSP (IAR, CMake, etc) to include any new files if this is the case.
> - Configure the module if required (typically the configuration is located in files suffixed with ``*_configuration.h``).
> - Now you should be able to build your BSP.

Reviewer: ```suggestion
- Update the toolchain build of your BSP (IAR, CMake, etc...) to include any new files if this is the case.
- Configure the C module if required (typically, the configuration is located in files suffixed with ``*_configuration.h``).
- Now you should be able to build your BSP firmware.
```
